test/bulk_schedule_test.cpp

```diff
@@ -48,3 +49,39 @@ TEST(bulk, bulk_transform) {
         EXPECT_EQ(i, output[i]);
     }
 }
+
+TEST(bulk, cancellation) {
+    unifex::single_thread_context ctx;
+    auto sched = ctx.get_scheduler();
+
+    const std::size_t count = 1000;
+
+    std::vector<int> output(count, 0);
+    // Cancel after two chunks
+    // For the serial implementation this will stop the third chunk onwards from
+    // being dispatched.
+    const std::size_t compare_index = unifex::bulk_cancellation_chunk_size*2 - 1;
```

Comment:
Is this `bulk_cancellation_chunk_size` something you expect to be part of the public interface of `bulk_schedule()`?

I imagine different schedulers might choose different strategies for `bulk_schedule()` which might have different behaviour for cancellation here.

Reply:
I expect so, yes. I wouldn't suggest standardising this, it should be implementation-defined.

Even the vectorisation length is implemetation-defined on analysis of the set_next chain.